### ISSUE TITLE
SSH connection issues on 6.0+

### ISSUE THREAD

**Issue by Aharic** [gravitational/teleport#6144]
### Description

**What happened**:

Customers have reported that they're unable to log into certain nodes at all after upgrading to 6.0+ So far customers who've reported the issue have reported it with Ubuntu 14.04 and Ubuntu 18.04 nodes. Neither ssh w/ port 3022, nor tsh, nor WebUI login work for affected nodes. The connections appear to establish (authentication looks fine on the auth server and node logs show connections starting) but then immediately disconnect due to a PTY read/write error. This behavior is not observed when nodes are downgraded back to 5.2.1

**What you expected to happen**:

Users should be able to connect to nodes via SSH and TSH on 6.0+

### Reproduction Steps

Still working on in-house reproduction, but not able to reproduce as of yet. Will update when successful.

### Server Details
- Teleport version (run `teleport version`): 6.0+
- Server OS (e.g. from `/etc/os-release`): Ubuntu (18.04 and 14.04)
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): Both cloud AWS EC2 instances and on-prem VMs.

### Client Details
- Tsh version (`tsh version`): 6.0+

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!

```
2021-03-24T14:58:45.335963-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [SSH:NODE]  Incoming connection 10.84.5.211:57038 -> 10.95.2.191:3022 vesion: SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.8. sshutils/server.go:443
2021-03-24T14:58:45.512091-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [KEEPALIVE] Starting keep-alive loop with with interval 5m0s and max count 3. srv/keepalive.go:65
2021-03-24T14:58:45.688556-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [NODE]      Handling request pty-req, want reply true. cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user regular/sshserver.go:1269
2021-03-24T14:58:45.688622-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [NODE]      Requested terminal "xterm-256color" of size {204 53} cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user srv/termhandlers.go:73
2021-03-24T14:58:45.688758-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU             Set permissions on /dev/pts/2 to 0:5 with mode -rw-------. srv/term.go:407
2021-03-24T14:58:45.688813-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [SESSION:N] Unable to update window size, no session found in context. srv/sess.go:389
2021-03-24T14:58:45.688893-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [NODE]      Handling request env, want reply false. cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user regular/sshserver.go:1269
2021-03-24T14:58:45.688943-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [NODE]      Handling request shell, want reply true. cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-usser regular/sshserver.go:1269
2021-03-24T14:58:45.699914-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [NODE]      Creating (interactive) session a0712c15-51c4-41a7-8d84-b6523e34ca53. cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user srv/sess.go:211
2021-03-24T14:58:45.699994-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [SESSION:N] Using async streamer for session a0712c15-51c4-41a7-8d84-b6523e34ca53. srv/sess.go:1043
2021-03-24T14:58:45.700511-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [SESSION:N] New party ServerContext(10.84.5.211:57038->10.95.2.191:3022, user=root, id=3) party(id=f13260a4-54be-4469-8607-3f533432b3b4) joined session: a0712c15-51c4-41a7-8d84-b6523e34ca53 srv/sess.go:1219
2021-03-24T14:58:45.700642-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU             Will join session a0712c15-51c4-41a7-8d84-b6523e34ca53 for SSH connection 10.84.5.211:57038. srv/ctx.go:435
2021-03-24T14:58:45.700801-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [SESSION:N] Starting poll and sync of terminal size to all parties. srv/sess.go:1161
2021-03-24T14:58:45.700859-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [AUDIT]     session.start addr.local:10.95.2.191:3022 addr.remote:10.84.5.211:57038 cluster_name:denmark1 code:T2000I ei:0 event:session.start login:root namespace:default server_addr:10.95.2.191:3022 server_hostname:lvs1-cph1 server_id:e901ad84-6575-4232-b097-e765c624be10 server_labels:map[ansible_groups:,consul-client,denmark1,denmark1-consul-client,denmark1-hardware,denmark1-lvs,denmark1-vault,europe,hardware,lvs,nsqd,vault, consul_services:hardware.syslog.metrics#012node.exporter#012nsq.exporter#012nsqd-http#012nsqd-tcp#012process.exporter#012teleport-ssh#012vault#012vault-exporter#012vault-statsd-exporter consul_wan_group:fusemail-production datacenter_id:12 datacenter_name:denmark1 hostname:lvs1-cph1 kernel:4.3.4-lxc region:DK role:lvs uptime:up 1 day, 22 hours, 22 minutes virtualization_type:hardware] sid:a0712c15-51c4-41a7-8d84-b6523e34ca53 size:204:53 time:2021-03-24T21:58:45.701Z uid:873d305e-004b-4cf0-96f7-701f1517f557 user:demo-user events/emitter.go:317
2021-03-24T14:58:45.726451-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [SESSION:N] **Copying from PTY to writer completed with error read /dev/ptmx: input/output error. srv/sess.go:788**
2021-03-24T14:58:45.726526-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [NODE]      Exec request ("/usr/local/bin/teleport") complete: 2 cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user regular/sshserver.go:1249
2021-03-24T14:58:45.726647-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [SESSION:N] Closing party f13260a4-54be-4469-8607-3f533432b3b4 srv/sess.go:1410
2021-03-24T14:58:45.726719-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [NODE]      Releasing associated resources - context has been closed. cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user srv/monitor.go:207
2021-03-24T14:58:45.726769-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU [SESSION:N] Sent session.join to 10.84.5.211:57038. srv/sess.go:290
2021-03-24T14:58:45.726833-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [NODE]      Removing party ServerContext(10.84.5.211:57038->10.95.2.191:3022, user=root, id=3) party(id=f13260a4-54be-4469-8607-3f533432b3b4) from session a0712c15-51c4-41a7-8d84-b6523e34ca53 cert:2021-03-24 21:36:10 -0700 PDT id:3 local:10.95.2.191:3022 login:root remote:10.84.5.211:57038 teleportUser:demo-user srv/sess.go:1083
2021-03-24T14:58:45.726905-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [SESSION:N] Session a0712c15-51c4-41a7-8d84-b6523e34ca53 will be garbage collected. srv/sess.go:323
2021-03-24T14:58:45.727027-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: INFO [SESSION:N] Closing session a0712c15-51c4-41a7-8d84-b6523e34ca53. srv/sess.go:632
2021-03-24T14:58:45.727078-07:00 lvs1-cph1 /usr/local/bin/teleport[16525]: DEBU             Closing session writer: session-recorder. srv/sess.go:1327
```

**Comment by xacrimon** [comment]
Couple notes from anecdotal issues with `/dev/ptmx´ which may be relevant for the one fixing this.
1. `/dev/ptmx` does not exist on all systems, on some distros it's `/dev/pts/ptmx`
2.  Some container systems don't invoke `udev` in a way that creates `/dev/ptmx`
3. `CONFIG_LEGACY_PTYS` needs to be enable in the kernel for `/dev/ptmx` to exist.
4. According to https://github.com/torvalds/linux/blob/v4.2/drivers/tty/n_tty.c#L2268-L2274 the kernel will return EIO when attempting to read or write to a master whose corresponding slave has closed.

**Comment by xacrimon** [comment]
Fixed in master with #6256 and in v6 with #6313
